### DISCUSSION TITLE
How to add words to user dictionary instead of workspace dictionary?

### DISCUSSION THREAD

**Discussion by jonas-frey** [valentjn/vscode-ltex#655] [Q&A]
I'm sure this has been asked  before but I've google around and haven't found an answer. 

If I do `ctrl+.` on an unknown word, I get a dropdown menu with the option `Add xyz to dictionary`. Selecting that puts the word in a wordlist in `PROJECTDIR/.vscode/`.

What if I want the word saved globally in `~/.vscode/` instead of per-workspace?

Thanks!

**Comment by valentjn** [comment]
Set [`ltex.configurationTarget`](https://valentjn.github.io/ltex/settings.html#ltexconfigurationtarget) for `"dictionary"` to `"user"`.

**Reply by jonas-frey** [reply]
Thanks! 

Now it writes the words into `~/.config/Code/User/settings.json`, into the `"ltex-dictionary"`-block. This block also contains the filename of a wordlist that I have compiled over the years. 

Is it possible to add the new entries directly to the external wordlist?

**Reply by valentjn** [reply]
You could use `"userExternalFile"` for `"dictionary"`. Then it writes to the first external file listed in the user configuration ([see docs](https://valentjn.github.io/ltex/vscode-ltex/setting-scopes-files.html#external-setting-files)).

**Reply by jonas-frey** [reply]
Could you tell me what you mean precisely by

> use `"userExternalFile"` for `"dictionary"`

?

I can't see any mention of `"userExternalFile"` in the linked page in the docs.

Thanks again!

**Reply by valentjn** [reply]
Instead of `"user"` as previously, use `"userExternalFile"` for the `"dictionary"` part of `ltex.configurationTarget`. Both values are documented in the linked documentation of my first reply.

The linked documentation in my second reply explains in detail what it does. Please read it fully instead of searching for a string.

> When executing a quick fix such as `Add to dictionary`, LT<sub>E</sub>X will do the following if the value of [`ltex.configurationTarget`](../settings.html#ltexconfigurationtarget) ends with `ExternalFile` (which it does by default): First, retrieve the list of the setting corresponding to the quick fix and to the language of the text in which the quick fix was executed. Then, pick the first entry that explicitly specifies an external file (i.e., the first entry starting with `:`). If there is no such entry, the implicit default path (see above) is used. The resulting external file is then appended with the entry to be added by the quick fix. If the file doesn't exist, it will be created, along with all necessary parent directories, if any are missing.
